src/Compilers/VisualBasic/Test/Emit/Emit/EmitMetadata.vb

```diff
@@ -973,5 +975,104 @@ End Class
+            Assert.False(emitResult.Success)
+            emitResult.Diagnostics.AssertTheseDiagnostics(<errors>
+BC30420: 'Sub Main' was not found in 'MyLib'.
+                                                          </errors>)
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76707")>
+        Public Sub EmitMetadataOnly_Exe_FriendMain_ExcludePrivateMembers()
+            CompileAndVerify(
+                <compilation>
+                    <file>
+Module Program
+    Friend Sub Main()
+    End Sub
+End Module
+                    </file>
+                </compilation>,
+                options:=TestOptions.ReleaseExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                emitOptions:=EmitOptions.Default.WithEmitMetadataOnly(True).WithIncludePrivateMembers(False),
+                symbolValidator:=Sub(m)
+                                     Assert.NotEqual(0, m.GetMetadata().Module.PEReaderOpt.PEHeaders.CorHeader.EntryPointTokenOrRelativeVirtualAddress)
+                                     Assert.NotNull(m.GlobalNamespace.GetMember(Of MethodSymbol)("Program.Main"))
+                                 End Sub
+            ).VerifyDiagnostics()
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76707")>
+        Public Sub ExcludePrivateMembers_FriendMain()
+            Using peStream As New MemoryStream()
+                Using metadataStream As New MemoryStream()
+                    Dim comp = CreateCompilation(
+                        <compilation>
+                            <file>
+Module Program
+    Friend Sub Main()
+    End Sub
+End Module
+                            </file>
+                        </compilation>,
+                        options:=TestOptions.ReleaseExe
+                    )
+                    Dim emitResult = comp.Emit(
+                        peStream:=peStream,
+                        metadataPEStream:=metadataStream,
+                        options:=EmitOptions.Default.WithIncludePrivateMembers(False))
+                    Assert.True(emitResult.Success)
+                    emitResult.Diagnostics.Verify()
+
+                    Verify(peStream)
+                    Verify(metadataStream)
+
+                    CompileAndVerify(comp).VerifyDiagnostics()
+                End Using
+            End Using
+        End Sub
+
+        Private Shared Sub Verify(stream As Stream)
+            stream.Position = 0
+            Assert.NotEqual(0, New PEHeaders(stream).CorHeader.EntryPointTokenOrRelativeVirtualAddress)
+
+            stream.Position = 0
+            Dim reference = AssemblyMetadata.CreateFromStream(stream).GetReference()
+            Dim comp = CreateCompilation("", references:={reference}, options:=TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All))
+            Assert.NotNull(comp.GetMember(Of MethodSymbol)("Program.Main") IsNot Nothing)
+        End Sub
```

Comment:
>End Sub [](http://example.com/codeflow?start=8&length=7)

It looks like VB doesn't support async Main. Still let's add a test for such scenario and observe an error.